packages/react-components/src/localization/LocalizationProvider.tsx

```diff
@@ -76,10 +76,74 @@ export type LocalizationProviderProps = {
  * @remarks Components will be provided localized strings in English (US) by default if this
  * provider is not used
  */
-export const LocalizationProvider = (props: LocalizationProviderProps): JSX.Element => {
+const InternalLocalizationProvider = (props: InternalLocalizationProviderProps): JSX.Element => {
   const { locale, children } = props;
   return <LocaleContext.Provider value={locale}>{children}</LocaleContext.Provider>;
 };
 
 /** React hook to access locale */
 export const useLocale = (): ComponentLocale => useContext(LocaleContext);
+
+interface SetLocaleContext {
+  setLocale: (locale?: string) => Promise<void>;
+}
+
+const SetLocaleContext = createContext<SetLocaleContext>({
```

Comment:
use camelCase: `setLocaleContext`, especially because it is currently shadowing the interface definition `SetLocaleContext`